[FIX] website: Do not fail if addon is missing in DB

Yajo opened the pull request:
If for any reason, any of the modules listed a few lines above is missing in the database, a user wouldn't be able to browse the website backend panel.

This can happen if an adventurous user deletes an uninstalled addon, or if some deployment system restricts the addons available for Odoo.

That's not a very common scenario, but in any case the fix is quite easy and here it is. It helps also downstream in case some additional apps are suggested to be installed using this same mechanism.

@Tecnativa TT21202




--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

pedrobaeza commented:
@JKE-be @qsm-odoo would you please consider this bugfix?

qsm-odoo commented on addons/website/static/src/xml/website.backend.xml:
```diff
@@ -37,7 +37,7 @@
-                    <t t-set="app_data" t-value="widget.dashboards_data.apps_data[app]"/>
+                    <t t-set="app_data" t-value="widget.dashboards_data.apps_data[app] || {'state': 'uninstallable'}"/>
                     <t t-set="app_dashboard" t-value="apps[app]"/>
                     <t t-if="vignette_nbr &lt; vignette_max &amp;&amp; widget.hidden_apps.indexOf(app_data['id']) == -1 &amp;&amp; app_data['state'] == 'uninstalled'">
```
Why not a simple `app_data && ...` in the condition L42 ? :)

Yajo commented:
Fixed:

- 72 chars max per commit message line.
- changed my simple fix for your simple fix. That line is now 188 chars long.

... your guidelines are weird :thinking:

Yajo commented:
Thanks! Interesting points. I didn't know some of them even when I've been contributing for years, though. :thinking: I guess this should be found in http://www.odoo.com/submit-pr

qsm-odoo commented:
I guess. Most are described here though: https://www.odoo.com/documentation/master/reference/guidelines.html#commit-message-structure even though the 72 characters one is not explicitly written (but the 50 characters one is)... our documentation... ;)
